Students endpoints

Deeathex opened the pull request.

Deeathex committed:
Student endpoints REST

Deeathex committed:
- toti studentii care au aplicat la un internship
- toti studentii care s-au abonat la o companie
- adaugarea unui student
- updatarea unui student
- un student sa aplice pentru un internship (AddApplication and updateApplication)

Deeathex committed:
Finished and tested:
- toti studentii care au aplicat la un internship
- toti studentii care s-au abonat la o companie
- adaugarea unui student
- updatarea unui student
- un student sa aplice pentru un internship (AddApplication and updateApplication)
- un student sa se subscrie la o companie (pt newsletters)

Deeathex committed:
Resolved conflicts

GhBogdan97 commented on DatabaseAccess/UOW/UnitOfWork.cs:
```diff
@@ -39,27 +41,42 @@ public IRepository<Student> StudentRepository
-        }
+		}
 
-        public IRepository<Post> PostRepository
```
asta nu ar trebui sters

GhBogdan97 commented on collective-project-backend/Controllers/StudentController.cs:
```diff
@@ -7,25 +7,162 @@
+		}
+
+		[HttpGet("getStudentsByInternshipId")]
+		public IActionResult GetStudentsByInternshipId(int InternshipId)
```
trebuie sa adaugi ruta pentru functie: [Route("internships/{internshipId}")] de ex, sa stie de unde vine InternshipId
si sa specifici ca rolul care are acces la functie e Compania

GhBogdan97 commented on collective-project-backend/Controllers/StudentController.cs:
```diff
@@ -7,25 +7,162 @@
+		{
+			//var InternshipId = Request.Cookies["InternshipId"];
+			var studentIds = _applicationService.GetStudentIdsByInternshipId(InternshipId);
+			var students = _studentService.GetAllStudents();
```
partea asta de logica ar trebui sa fie in service, si sa devina tot doar un IList<Student> = _studentService.StudentsByInternshipID(studentIds) de ex.

GhBogdan97 commented on collective-project-backend/Controllers/StudentController.cs:
```diff
@@ -7,25 +7,162 @@
+					}
+				}
+			}
+			return Ok(studentsByInternshipId);
```
ceea ce returnezi ar trebui sa fie o lista de ViewModels, nu o lista cu obiecte din baza de date, eventual ar fi util sa discuti cu cineva de pe frontend si sa vezi ce ar avea nevoie cand afiseaza studentii de aici (ma gandesc ca au nevoie doar de nume, prenume, +- 2 fielduri)

GhBogdan97 commented on collective-project-backend/Controllers/StudentController.cs:
```diff
@@ -7,25 +7,162 @@
+		}
+
+		[HttpGet("GetStudentsByCompanyId")]
+		public IActionResult GetStudentsByCompanyId(int CompanyId)
```
la fel, adaugate ruta + rol

GhBogdan97 commented on collective-project-backend/Controllers/StudentController.cs:
```diff
@@ -7,25 +7,162 @@
+			var studentIds = _subscriptionService.GetStudentIdsByCompanyId(CompanyId);
+			var students = _studentService.GetAllStudents();
+			IList<Student> studentsByCompanyId = new List<Student>();
+			foreach (Student student in students)
```
la fel, mutata logica in service

GhBogdan97 commented on collective-project-backend/Controllers/StudentController.cs:
```diff
@@ -7,25 +7,162 @@
+					}
+				}
+			}
+			return Ok(studentsByCompanyId);
```
la fel, returnata o lista cu ViewModels

GhBogdan97 commented on collective-project-backend/Controllers/StudentController.cs:
```diff
@@ -7,25 +7,162 @@
+		}
+
+		[HttpPost("updateStudent")]
+		public IActionResult UpdateStudent([FromBody] Student student)
```
obiectul primit in body ar trebui sa fie un ViewModel (care sa contina doar fieldurile ce pot fi updatate), nu tot obiectul din db
ar trebui adaugat rolul care poate face update unui student

GhBogdan97 commented on collective-project-backend/Controllers/StudentController.cs:
```diff
@@ -7,25 +7,162 @@
+		}
+
+		[HttpPost("addStudent")]
+		public IActionResult AddStudent([FromBody] Student student)
```
obiectul primit in body ar trebui sa fie un ViewModel (de exemplu id-ul nu e setat de noi, ci e setat automat la adaugarea in db), nu tot obiectul din db

GhBogdan97 commented on collective-project-backend/Controllers/StudentController.cs:
```diff
@@ -7,25 +7,162 @@
+		}
+
+		[HttpPost("addApplication")]
+		public IActionResult AddApplication([FromBody] Application application)
```
ar trebui pus rolul care poate aplica.
obiectul primit in body ar trebui sa fie un ViewModel

GhBogdan97 commented on collective-project-backend/Controllers/StudentController.cs:
```diff
@@ -7,25 +7,162 @@
+		}
+
+		[HttpPost("updateApplication")]
+		public IActionResult UpdateApplication([FromBody] Application application)
```
obiectul primit in body ar trebui sa fie un ViewModel
ar trebui pus rolul care poate face update

GhBogdan97 commented on collective-project-backend/Controllers/StudentController.cs:
```diff
@@ -7,25 +7,162 @@
+		}
+
+		[HttpPost("addSubscription")]
+		public IActionResult AddSubscription([FromBody] Subscription Subscription)
```
obiectul primit in body ar trebui sa fie un ViewModel
ar trebui pus rolul care poate face subscribe

GhBogdan97 commented on collective-project-backend/Controllers/StudentController.cs:
```diff
@@ -7,25 +7,162 @@
+		}
+
+		[HttpPost("updateSubscription")]
+		public IActionResult UpdateSubscription([FromBody] Subscription Subscription)
```
obiectul primit in body ar trebui sa fie un ViewModel
ar trebui pus rolul care poate face update

GhBogdan97 reviewed (dismissed):
sa faci te rog si merge din dev sa poti rezolva conflictele + comentariile

Deeathex committed:
Resolved conflicts

Deeathex committed:
Refactored services and controllers + created mappers and model views

Deeathex committed:
Refactored register + get posts for internship added

Deeathex committed:
Added endpoints:
- check is student applied for an internship (exists application)
- check if exists rating for a student to an internship
- fixed at update application - if a student is finally admitted to an internship, his application for all other applications would be marked as refused
- also made new Controllers, Services, etc

Deeathex committed:
After merge from dev to StudentEndpoints
